### PR TITLE
lib/ec_glob: plug leak of nums utarray

### PR DESCRIPTION
**Pull request by chergert**
Static analysis found a number of leaks through the use of the convenience macros.

### CODE REVIEW

File: src/lib/ec_glob.c [hunk 1]
```diff
@@ -53,16 +53,20 @@ static const UT_icd ut_int_pair_icd = {sizeof(int_pair),NULL,NULL,NULL};
 #define STRING_CAT(p, string, end)  do {    \
     size_t string_len = strlen(string); \
     assert(end > p); \
-    if (string_len >= (size_t)(end - p)) \
+    if (string_len >= (size_t)(end - p)) { \
+        utarray_free(nums); \
```
Comment (xuhdev) [inline src/lib/ec_glob.c:57]: I would say `goto cleanup` is more preferable, which also frees any pcre2 handles if present.

File: src/lib/ec_glob.c [hunk 2]
```diff
@@ -53,16 +53,20 @@ static const UT_icd ut_int_pair_icd = {sizeof(int_pair),NULL,NULL,NULL};
...
+    } \
     strcat(p, string); \
     p += string_len; \
 } while(0)
 
 /* safely add a char to a string then move the pointer to the end */
 #define ADD_CHAR(string, new_chr, end)  do {    \
-    if (string + 1 >= end) \
+    if (string + 1 >= end) { \
+        utarray_free(nums); \
```
Comment (xuhdev) [inline src/lib/ec_glob.c:67]: same as above

**Review by xuhdev — COMMENTED** [review]
Thanks for the contribution!

**Comment by chergert** [comment]
SGTM

File: src/lib/ec_glob.c [hunk 3]
```diff
@@ -54,15 +54,15 @@ static const UT_icd ut_int_pair_icd = {sizeof(int_pair),NULL,NULL,NULL};
     size_t string_len = strlen(string); \
     assert(end > p); \
     if (string_len >= (size_t)(end - p)) \
-        return -1; \
```
Comment (xuhdev) [inline src/lib/ec_glob.c:57]: Sorry, forgot to say that you need to add the following above `goto`:

```
ret = -1;
```

File: src/lib/ec_glob.c [hunk 4]
```diff
@@ -54,15 +54,15 @@ static const UT_icd ut_int_pair_icd = {sizeof(int_pair),NULL,NULL,NULL};
...
+        goto cleanup; \
     strcat(p, string); \
     p += string_len; \
 } while(0)
 
 /* safely add a char to a string then move the pointer to the end */
 #define ADD_CHAR(string, new_chr, end)  do {    \
     if (string + 1 >= end) \
-        return -1; \
+        goto cleanup; \
```
Comment (xuhdev) [inline src/lib/ec_glob.c:65]: Same as above.

**Review by xuhdev — APPROVED** [review]
Thanks!

**Pull request merged**